src/Sulu/Component/Webspace/Manager/WebspaceManager.php

```diff
@@ -288,7 +291,7 @@ public function findUrlByResourceLocator(
 
         $this->portalUrlCache[$webspaceKey][$domain][$environment][$languageCode] = $portalUrl;
 
-        if (!$portalUrl) {
+        if (!$portalUrl || $resourceLocator === null) {
```

Comment:
how it can be null if it overwritten above?

```
        if (null === $resourceLocator) {
            $resourceLocator = '/';
        }
```

Reply:
Yes I first had the intention to catch it and return `null`, but the current implementation falls back to an empty string which will return the homepage, so I forgot to revert this part